### ISSUE TITLE
cannot read property "state" of null

### ISSUE THREAD

**Issue by jptrsn** [AYapejian/node-red-contrib-home-assistant#5]
It looks like there's an unhandled exception state that's causing node-red to crash on my installation of HassIO.

```
6 Nov 23:33:51 - TypeError: Cannot read property 'state' of null
    at Object.onIncomingMessage (/data/node_modules/node-red-contrib-home-assistant/node-server-state-changed/node-server-state-changed.js:46:147)
    at HaEvents.onStateChanged (/data/node_modules/node-red-contrib-home-assistant/node-server-state-changed/node-server-state-changed.js:54:43)
    at emitOne (events.js:101:20)
    at HaEvents.emit (events.js:188:7)
    at HaEvents.onClientMessage (/data/node_modules/node-home-assistant/lib/ha-events.js:57:18)
    at EventSource.HaEvents.client.on (/data/node_modules/node-home-assistant/lib/ha-events.js:22:49)
    at emitOne (events.js:96:13)
    at EventSource.emit (events.js:188:7)
    at _emit (/data/node_modules/eventsource/lib/eventsource.js:186:17)
    at parseEventStreamLine (/data/node_modules/eventsource/lib/eventsource.js:201:9)
    at IncomingMessage.<anonymous> (/data/node_modules/eventsource/lib/eventsource.js:164:11)
    at emitOne (events.js:96:13)
    at IncomingMessage.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at IncomingMessage.Readable.push (_stream_readable.js:134:10)
    at HTTPParser.parserOnBody (_http_common.js:123:22)
npm info lifecycle node-red-docker@1.0.0~start: Failed to exec start script
npm ERR! Linux 4.4.50
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start" "--" "--userDir" "/data"
npm ERR! node v6.11.0
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! node-red-docker@1.0.0 start: `node $NODE_OPTIONS node_modules/node-red/red.js -v $FLOWS "--userDir" "/data"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-red-docker@1.0.0 start script 'node $NODE_OPTIONS node_modules/node-red/red.js -v $FLOWS "--userDir" "/data"'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the node-red-docker package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node $NODE_OPTIONS node_modules/node-red/red.js -v $FLOWS "--userDir" "/data"
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs node-red-docker
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls node-red-docker
npm ERR! There is likely additional logging output above.
npm ERR! Please include the following file with any support request:
npm ERR!     /usr/src/node-red/npm-debug.log
```

**Comment by gregg098** [comment]
I'm having the same issue. I have to manually start NodeRed after any restart/reboot of HassIO.

**Comment by tehniemer** [comment]
I've gotten this same error also.

**Comment by AYapejian** [comment]
Thanks for the report.  I don't have much of an update, but just wanted to let everyone know I haven't abandoned this project.  This repo is next on my priority list after I get to a good stopping point on my current project.  

I actually did a rather large rewrite of the lib this addon uses, [node-home-assistant](https://github.com/AYapejian/node-home-assistant) I'll have to update to accommodate those changes and then address bugs and requests after that.  I'll attempt to keep things backwards compatible if I can.  Too many projects, not enough time I'm afraid

**Comment by jamesgol** [comment]
Noticed this hit me last night too, it doesn't seem to happen every time I've restarted homeassistant.

Since the earlier code in onIncomingMessage checks if (!event || !event.new_state) perhaps its the event.old_state that is null that the code is trying to pass to the node.debug() function.

I'll try to change my local copy and see if I can force this to reproduce itself.


```
22 Jan 22:42:51 - [error] [server-state-changed:2ad5c29d.ac8dfe] Connection error occured with the home-assistant server: {"type":"error"}

22 Jan 22:42:53 - [error] [server-state-changed:2ad5c29d.ac8dfe] Connection error occured with the home-assistant server: {"type":"error"}

22 Jan 22:42:55 - [warn] [server-state-changed:2ad5c29d.ac8dfe] [object Object]
22 Jan 22:42:55 - [warn] [server-state-changed:2ad5c29d.ac8dfe] [object Object]
22 Jan 22:42:55 - [red] Uncaught Exception:
22 Jan 22:42:55 - TypeError: Cannot read property 'state' of null
at Object.onIncomingMessage (/data/node_modules/node-red-contrib-home-assistant/node-server-state-changed/node-server-state-changed.js:46:147)
at HaEvents.onStateChanged (/data/node_modules/node-red-contrib-home-assistant/node-server-state-changed/node-server-state-changed.js:54:43)
at emitOne (events.js:101:20)
at HaEvents.emit (events.js:188:7)
at HaEvents.onClientMessage (/data/node_modules/node-home-assistant/lib/ha-events.js:57:18)
at EventSource.client.on (/data/node_modules/node-home-assistant/lib/ha-events.js:96:49)
at emitOne (events.js:96:13)
at EventSource.emit (events.js:188:7)
at _emit (/data/node_modules/eventsource/lib/eventsource.js:186:17)
at parseEventStreamLine (/data/node_modules/eventsource/lib/eventsource.js:201:9)
at IncomingMessage.<anonymous> (/data/node_modules/eventsource/lib/eventsource.js:164:11)
at emitOne (events.js:96:13)
at IncomingMessage.emit (events.js:188:7)
at readableAddChunk (_stream_readable.js:176:18)
at IncomingMessage.Readable.push (_stream_readable.js:134:10)
at HTTPParser.parserOnBody (_http_common.js:123:22)
at Socket.socketOnData (_http_client.js:363:20)
at emitOne (events.js:96:13)
at Socket.emit (events.js:188:7)
at readableAddChunk (_stream_readable.js:176:18)
at Socket.Readable.push (_stream_readable.js:134:10)
at TCP.onread (net.js:547:20)

npm ERR! Linux 4.14.13-unRAID
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start" "--" "--userDir" "/data"
npm ERR! node v6.12.0
npm ERR! npm v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! node-red-docker@1.0.0 start: `node $NODE_OPTIONS node_modules/node-red/red.js -v $FLOWS "--userDir" "/data"`
npm ERR! Exit status 1
```

**Comment by jmesberg** [comment]
AYapejian, this is probably not good github protocol, but I just wanted to let you know that I absolutely love what you've done here. I installed Node-RED and your repository last night and in no time had everything working and had already built 3-4 useful automations. I like this much better than Appdaemon (though that has its place for sure) and WAY better than messing with automations.yaml! Great work. I hope you keep the project active, and that others contribute to it.

**Comment by jptrsn** [comment]
I've had a bit of luck suppressing the crashes by applying an entity id filter to the state changed node. This appears to work, except for occasions where the new_state is null. This happens when you create then dismiss a persistent notification in HA - upon dismissal, the new_state is null. When a persistent notification is dismissed, I get an error logged from every state change node I'm using (despite their having entity id filters to prevent this from happening).

**Comment by AYapejian** [comment]
Merged to master and published as 0.2.1 to npm
